contrib_bots/lib/hangman.py

```diff
@@ -0,0 +1,161 @@
+    bot is subscribed to using the following simple commands:
+    - @hangman new: This command starts a new Hangman game.
+    - @hangman guess <letter>: This command lets the user
+    guess a letter for the current Hangman game.
+    - @hangman stop: This command stops the current Hangman game.
+    - @hangman help: This command gives users information
+    about the Hangman bot.
+    '''
+
+    def triage_message(self, message, client):
+        # Return True if we want to (possibly) respond to this message
+        original_content = message['content']
+        is_hangman_command = original_content.lower().startswith('@hangman ')
+
+        return is_hangman_command
+
+    def get_new_word(bot_self):
+        link = "http://www.setgetgo.com/randomword/get.php"
+        if not hasattr(bot_self, 'currentword'):
+            try:
+                bot_self.currentword = requests.get(link).text
+                bot_self.output = "To start guessing letters, type `@hangman" \
+                    " guess <letter>`. Good luck!\n\n```" \
+                    "" + "  ".join("_" * len(bot_self.currentword)) + "```"
+                bot_self.stage = 0
+                bot_self.guessed = []
+            except:
+                bot_self.output = "Could not generate a word at this moment." \
+                    " Try again later."
+        else:
+            if len(bot_self.currentword) == 0:
+                bot_self.currentword = requests.get(link).text
+                bot_self.output = "To start guessing letters, " \
+                    "type `@hangman guess <letter>`. Good luck!\n\n```" \
+                    "" + "  ".join("_" * len(bot_self.currentword)) + "```"
+                bot_self.stage = 0
+                bot_self.guessed = []
+            else:
+                bot_self.output = "There is already a game of hangman " \
+                    "active! To stop the current game, type `@hangman stop`."
+
+    def process_guess(bot_self, message):
+        end = ""
+        bot_self.ended = False
+        split = message.strip().split(" ")
+        if len(split) == 2:
+            bot_self.output = "You forgot to guess a letter! " \
+                "To guess again, type `@hangman guess <letter>`."
+        elif len(split) > 3:
+            bot_self.output = "You are supposed to guess one letter " \
+                "at a time, not multiple words at once!" \
+                " To guess again, type `@hangman guess <letter>`."
+        elif len(split[2]) > 1:
+            bot_self.output = "You are supposed to guess a letter, " \
+                "not a word! To guess again, type `@hangman guess <letter>`."
+        elif split[2] in bot_self.guessed:
+            bot_self.output = "You have already guessed the letter " \
+                "" + split[2] + "! To guess again, type " \
+                "`@hangman guess <letter>`."
+        elif split[2] not in "abcdefghijklmnopqrstuvwxyz":
```

Comment:
could you use `string` module constant?

that should be something like:
`import string`
`string.alphabet` or something similar - I'm almost sure that there is constant with lowercase alphabet too.